zitadel/v2/org_idp_github/model.go

```diff
@@ -0,0 +1,101 @@
+package org_idp_github
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/zitadel/zitadel-go/v2/pkg/client/zitadel/idp"
+	"github.com/zitadel/zitadel-go/v2/pkg/client/zitadel/management"
+)
+
+type tfModel struct {
+	// ID is never null or empty, so we can use the native type
+	ID string `tfsdk:"id"`
+	// OrgID is never null or empty, so we can use the native type
+	OrgID             string       `tfsdk:"org_id"`
+	Name              types.String `tfsdk:"name"`
+	ClientID          types.String `tfsdk:"client_id"`
+	Scopes            []string     `tfsdk:"scopes"`
+	IsLinkingAllowed  types.Bool   `tfsdk:"is_linking_allowed"`
+	IsCreationAllowed types.Bool   `tfsdk:"is_creation_allowed"`
+	IsAutoCreation    types.Bool   `tfsdk:"is_auto_creation"`
+	IsAutoUpdate      types.Bool   `tfsdk:"is_auto_update"`
+}
+
+// We have to duplicate the fields until the framework supports embedding
+// https://github.com/hashicorp/terraform-plugin-framework/issues/242
+type tfModelSensitive struct {
+	// ID is never null or empty, so we can use the native type
+	ID string `tfsdk:"id"`
+	// OrgID is never null or empty, so we can use the native type
+	OrgID             string       `tfsdk:"org_id"`
+	Name              types.String `tfsdk:"name"`
+	ClientID          types.String `tfsdk:"client_id"`
+	Scopes            []string     `tfsdk:"scopes"`
+	IsLinkingAllowed  types.Bool   `tfsdk:"is_linking_allowed"`
+	IsCreationAllowed types.Bool   `tfsdk:"is_creation_allowed"`
+	IsAutoCreation    types.Bool   `tfsdk:"is_auto_creation"`
+	IsAutoUpdate      types.Bool   `tfsdk:"is_auto_update"`
+	ClientSecret      types.String `tfsdk:"client_secret"`
+}
+
+func (m *tfModelSensitive) toPbAddGithubProviderRequest() *management.AddGitHubProviderRequest {
+	return &management.AddGitHubProviderRequest{
+		Name:            m.Name.ValueString(),
+		ClientId:        m.ClientID.ValueString(),
+		ClientSecret:    m.ClientSecret.ValueString(),
+		Scopes:          m.Scopes,
+		ProviderOptions: m.toPbIdpOptions(),
+	}
+}
+
+func (m *tfModelSensitive) toPbUpdateGithubProviderRequest() *management.UpdateGitHubProviderRequest {
+	return &management.UpdateGitHubProviderRequest{
+		Id:              m.ID,
```

Comment:
shouldn't it also be 
```suggestion
		Id:              m.ID.ValueString(),
```